packages/panzer/disc-fe/src/Panzer_Workset.cpp

```diff
@@ -182,6 +185,50 @@ void WorksetDetails::setupNeeds(Teuchos::RCP<const shards::CellTopology> cell_to
 
   }
 
+  // Create the point terms and basis-integrator pairs
+  for(const panzer::PointDescriptor & point_description : point_descriptors){
+
+    // get the generaotr, and build some points from a topology
+    auto points = point_description.getGenerator().getPoints(*cell_topology);
+
+    // Create and store integration rule
+    Teuchos::RCP<panzer::PointRule> pr = Teuchos::rcp(new panzer::PointRule(point_description.getType(), 
```

Comment:
I believe IntegrationRule can be constructed from an IntegrationDescriptor, and a PureBasis can be constructed from a BasisDescriptor. Should we do the same thing here? Basically, it would do the 'getPoints' call internally, and only require num_cells and num_topology (and num faces?).